apps/src/studio/studio.js

```diff
@@ -4767,6 +4811,15 @@ Studio.checkRequiredForSuccess = function() {
   return { exists: true, achieved: true, message: null };
 };
 
+/**
+ * Trigger a manual failure, which stops the interpreter and ends the level
+ * prematurely.
+ * @param {string} [message] optional failure message text
+ */
+Studio.fail = function (message) {
```

Comment:
A comment may be needed here. As currently implemented, this function may not as reliably in a couple of cases:
* When in the middle of a type loop of interpreter code, I think we will continue to execute a bunch of the student's code for a while before checking for `Studio.midExecutionFailure` at the end of the tick.
* For blockly mode, where there is no interpreter, we have the same issue.